controllers/jujucluster_controller.go

```diff
@@ -865,23 +873,31 @@ func createApplicationsIfNeeded(ctx context.Context, jujuCluster *infrastructure
 			ApplicationName: "kubernetes-control-plane",
 			ModelUUID:       modelUUID,
 			CharmName:       "kubernetes-control-plane",
-			CharmChannel:    "1.26/stable",
+			CharmChannel:    "1.27/edge",
 			CharmBase:       "ubuntu@22.04",
 			Units:           0,
```

Comment:
```suggestion
			Units:           0,
			Config: map[string]interface{}{
				"ignore-missing-cni": true,
				"enable-metrics":     false,
				"dns-provider":       "none",
			},
```

Reply:
kcp charm has been updated to ignore missing CNI, together with your changes for ignoring missing workers, the charm will now reach active/idle once units are deployed. 

note that metrics and dns must be disabled as well, as these pods cannot start without a CNI.

Reply:
now that kcp is going active/idle, we could wait for this condition in the machine controller and once its met, set ready status.